sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonService.java

```diff
@@ -397,10 +401,28 @@ public Location getAddressByPersonId(long personId) {
 		return result;
 	}
 
-	public Predicate buildSimilarityCriteriaFilter(PersonSimilarityCriteria criteria, CriteriaBuilder cb, From<Person, Person> personFrom) {
+	public Predicate buildSimilarityCriteriaFilter(PersonSimilarityCriteria criteria, CriteriaBuilder cb, From<?, Person> personFrom) {
 
 		Predicate filter = null;
 
+		if (criteria.getFirstName() != null && criteria.getLastName() != null) {
+			Expression<String> nameExpr = cb.concat(personFrom.get(Person.FIRST_NAME), " ");
+			nameExpr = cb.concat(nameExpr, personFrom.get(Person.LAST_NAME));
+
+			Expression<String> nameInvertedExpr = cb.concat(personFrom.get(Person.LAST_NAME), " ");
+			nameInvertedExpr = cb.concat(nameInvertedExpr, personFrom.get(Person.FIRST_NAME));
+
+			String name = criteria.getFirstName() + " " + criteria.getLastName();
+			String nameInverted = criteria.getLastName() + " " + criteria.getFirstName();
+
+			double nameSimilarityThreshold = Math.max(0, configFacade.getNameSimilarityThreshold() - 0.2);
```

Comment:
I've added 0.2 less in the query to speed it up, and I also kept the java code version with the original threshold so the users will see the same result.

I did like this because I'm not sure the similarity function in the query is the same as the one used in java code.